### DISCUSSION TITLE
Back / Navigation Button

### DISCUSSION THREAD

**Discussion by iStump** [music-assistant/support#2736] [Feature requests and ideas]
Hello, I am basically very satisfied with Music Assistant, but I would like to have a back or navigation button.
Since I use my iPad in Kiosk mode, I can no longer get out of Music Assitant unless I quit Home Assistant. Or am I missing something?

**Comment by mrgrlscz** [comment]
I think you're not missing anything.... Or I am too!
I also need this feature.

**Comment by MankiniChykan** [comment]
Up vote

**Reply by OzGav** [reply]
Just checking did you press the upvote button on the first post?

**Comment by laurens362** [comment]
Upvote

**Comment by rdehuyss** [comment]
That would make it even more awesome.

**Comment by DarthRidar** [comment]
I have the same problem. There's no back button on my wall tablet. The navigation bar is hidden.

**Comment by deadgemini-web** [comment]
Upvoted. This almost seems like an oversight, in my opinion. Pretty basic feature, I think. Would be very cool if we could just add custom buttons to the MA sidebar ourselves.

**Comment by AkoreJepade** [comment]
Much needed! Upvoted!

**Comment by Erwin-M90** [comment]
It`s so sad that we didn`t see this feature in the recent 2.8.0 release. This is a much needed feature to get back to the dashboard for people using full kiosk mode.

**Comment by marcelveldt** [comment]
I'm running fully kiosk browser myself as well and don't see any issues. How are you all running MA and opening it?

**Reply by Erwin-M90** [reply]
Thank you for your response Marcel! Much appreciated.

I run Home Assistant on a tablet in full kiosk mode using the: 
HACS "kiosk-mode" from NemesisRE
http://homeassistant.local:8123/hacs/repository/497319128

I have no header and sidebar. Because of this. If I open music assistant directly via a button on my dashboard there is no way I can back to my dashboard. I currently created a hacky solution where I open music assistant in a custom iFrame on a dashboard page via ingress in full screen. I overlayed the page with a custom home button. But this is resulting in an issue with scrolling and the music assistant native menu bar. We would like to just have an option to add a custom menu button so we can link back to our dashboard.

In v2.8.0 I also seem to have an issue with Music Assistant menu not closing when clicking the menu button. Maybe this is a compatibility issue with the Kiosk mode.

**Reply by Erwin-M90** [reply]
How are you getting back from Music Assistant to your dashboard without header and sidebar?

**Reply by marcelveldt** [reply]
I'm using default HAOS Ingress panel so I can simply go back from the header to open the HA menu.
Be aware that when MA is opened through HA, it is HA that controls the back control and menu - making this rather a feature request for HA than for MA.

BTW: We are working on a feature where MA taps in the HA menu and you can basically open/close the HA menu from within HA so that might be the solution you're looking for

**Reply by Erwin-M90** [reply]
That`s what we are currently avoiding with kiosk mode. I use this tablet with Home Assistant pinned so people can access it without needing to unlock the tablet. I don`t want people to access the normal menu even though I removed a lot of the settings there. I created my own bottom navigation bar. I don`t need my own navigation bar inside Music Assistant but at minimum I need a way back to my dashboard and that`s why we would like a custom navigation button to get back. If you for some reason can`t give us that maybe it`s a solution to create a buildin button that you can activate that links to the default dashboard that is set in Home Asisstant?

**Reply by Erwin-M90** [reply]
@marcelveldt any update around this?

**Reply by marcelveldt** [reply]
No. This either needs to be fixed in the HA app or maybe wait for the feature on our end where you can open the HA menu from our MA menu which is planned somewhere between now and 2 months.

Note that this works by default. It's the (non standard) fact that you are hiding the HA menubutton that this broke for you.

**Reply by Erwin-M90** [reply]
Alright, that is really unfortunate that you don`t support full kiosk mode and are not planning to. I think a lot of people are using this. I guess that`s why a lot of people are using other media players instead of music assistant. To me it seems like an extra configurable navigation button should not be that big of a deal but I could be wrong. The solution of tapping into the native HA menu is not a solution in a full kiosk mode setup. At least thank you for replying to my messages.

**Reply by marcelveldt** [reply]
You are not getting my point. This has nothing to do with MA at all but with the HA app / frontend which is hosting MA in your case

**Reply by Erwin-M90** [reply]
No, Im not really getting your point. Because even if it is an issue that is created by HA itself the users of your app run into it. I think a big part of the HA community is using kiosk mode. I think its quite simple to work around it and add a simple optional navigation button that is deactivated by default. This ticket has quite some upvotes and I do not really understand why you would not add this. Also the option that you propose with tapping into the default HA menu is not really an option for this group of people because they hide it for a reason. Hopefully you reconsider this small feature.

**Comment by DarthRidar** [comment]
Will we get a back button or not? That would be great.
